### PR TITLE
[13.x] Fix Factory::raw() with a count of zero

### PR DESCRIPTION
**Pull request by OpadaAlzaiede**
### **The Problem:**
  The Factory raw() method returns a non-zero list in the following scenarios:
  
  
  ```
  // This returns a list of 2 elements while the expected is an empty list
  UserFactory::new()->count(0)->raw()
  ```
  ```
  // This returns a list of 5 elements while the expected is an empty list
  UserFactory::new()->count(-3)->raw();
  ```
  
  ### Why?
  The range() function is behaving this way:
  ```
  range(1,0) -> [1, 0]
  range(1,-3) -> [1, 0, -1, -2, -3]
  ```
  
  ### The Solution:
  Add this check in the raw() method implementation:
  ```
  if ($this->count < 1) {
        return [];
  }
  ```
  
  The Factory make() method already includes this condition.
  
  
  ### Test:
  vendor\bin\phpunit tests\Database\DatabaseEloquentFactoryTest.php --filter test_model_attributes_can_be_created_with_a_count_of_zero

### CODE REVIEW

**Review by crynobone — APPROVED** [review]

File: tests/Database/DatabaseEloquentFactoryTest.php [hunk 1]
```diff
@@ -269,6 +269,13 @@ public function test_multiple_model_attributes_can_be_created()
         $this->assertCount(10, $posts);
     }
 
+    public function test_model_attributes_can_be_created_with_a_count_of_zero()
+    {
+        $posts = FactoryTestPostFactory::new()->count(0)->raw();
```
Comment (crynobone) [inline tests/Database/DatabaseEloquentFactoryTest.php:274]: might as well use `TestWith` and also test negative values
